src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerExecutor.cs

```diff
@@ -292,15 +294,19 @@ public void ExecuteSuppressionAction(DiagnosticSuppressor suppressor, ImmutableA
             cancellationToken.ThrowIfCancellationRequested();
 
             var supportedSuppressions = _analyzerManager.GetSupportedSuppressionDescriptors(suppressor, this, cancellationToken);
-            Func<SuppressionDescriptor, bool> isSupportedSuppression = supportedSuppressions.Contains;
-            Action<SuppressionAnalysisContext> action = suppressor.ReportSuppressions;
+
+            using var _ = PooledDelegates.GetPooledFunction(
+                static (d, supportedSuppressions) => supportedSuppressions.Contains(d),
```

Comment:
nit: this would have helped me understand what this code was doing a little faster.

```suggestion
                static (descriptor, supportedSuppressions) => supportedSuppressions.Contains(descriptor),
```

I think the fact that the resulting function signature is only hinted at in the `out` argument, which comes last, made the first few of these I read a bit slow to grasp.

Reply:
If you decide to make a change here, feel free to do so in follow-up PR. Thanks

Reply:
sure.  let me do that in followup.  i agree that the single letters are not always great.